### PR TITLE
KEP-1606 - don't forward messages with invalid timestamps

### PR DESCRIPTION
**Pull request by paularchard**
Fix for issue on testnet where bad client timestamp was causing viewchanges to occur.

### CODE REVIEW

File: pbft/pbft.cpp [hunk 1]
```diff
@@ -303,6 +303,25 @@ pbft::setup_request_operation(const bzn_envelope& request_env, const bzn::hash_t
 void
 pbft::handle_request(const bzn_envelope& request_env, const std::shared_ptr<session_base>& session)
 {
+    if (request_env.timestamp() < (this->now() - MAX_REQUEST_AGE_MS) || request_env.timestamp() > (this->now() + MAX_REQUEST_AGE_MS))
+    {
+        database_msg req;
+        if (session && req.ParseFromString(request_env.database_msg()))
+        {
+            database_response resp;
+            resp.set_allocated_header(new database_header(req.header()));
```
Comment (ebruck) [inline pbft/pbft.cpp]: you don't need to do this.
resp.mutable_header()->xxxx will allocate automatically. We no longer use set_allocated in the daemon.

File: pbft/pbft.cpp [hunk 2]
```diff
@@ -303,6 +303,25 @@ pbft::setup_request_operation(const bzn_envelope& request_env, const bzn::hash_t
...
+            auto err = new database_error;
+            err->set_message(TIMESTAMP_ERROR_MSG);
+            resp.set_allocated_error(err);
```
Comment (ebruck) [inline pbft/pbft.cpp]: same here

File: pbft/pbft.cpp (hunk 1 above)
Comment (paularchard) [reply pbft/pbft.cpp]: How do I set the entire header from another header?

Comment (paularchard) [reply pbft/pbft.cpp]: done

File: pbft/pbft.cpp (hunk 2 above)
Comment (paularchard) [reply pbft/pbft.cpp]: done

**Comment by paularchard** [comment]
Also added a send error response to the "too large request" case as that would also cause a re-transmit loop otherwise.

File: pbft/pbft.cpp (hunk 1 above)
Comment (ebruck) [reply pbft/pbft.cpp]: *response.mutable_header() = request.header();

**Pull request merged**